lib/utils/ensureExists.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const fse = require('fs-extra');
+const fs = require('fs');
+const path = require('path');
+
+module.exports = async (filename, generate) => {
+  const cacheDir = path.dirname(filename);
```

Comment:
I would keep `cacheDir` resolution out of this file. It doesn't seem to belong in its scope.

`ensureArticate`  has `cachePath` resolved upfront, and it may just return it after calling `ensureExists`. I think such approach will be cleaner and less confusing.

Reply:
I still need to resolve `cacheDir`, even when not returning it as it's the expected signature of `generate` method. Do you suggest to also refactor that and corresponding `generate` methods?

Reply:
Indeed, I forgot about that. Sorry for confusion.

I see you've made some updates, let's keep it like that 👍